woodwork/statistics_utils.py

```diff
@@ -210,18 +219,35 @@ def _get_mutual_information_dict(dataframe, num_bins=10, nrows=None, include_ind
     # calculate mutual info for all pairs of columns
     mutual_info = []
     col_names = data.columns.to_list()
+
+    n = len(col_names)
+    total_loops = (n * n + n) / 2
+
+    def update_progress(start_time, increment, current, total, progress_callback):
+        new_progress = current + increment
+        if progress_callback is not None:
+            elapsed_time = timer() - start_time
+            progress_callback((increment / total) * 100, (new_progress / total) * 100, elapsed_time)
+
+        return new_progress
+
+    current_progress = update_progress(start_time, 0, 0, total_loops, progress_callback)
```

Comment:
Depending on how long `_replace_nans_for_mutual_info` and `_make_categorical_for_mutual_info` take, this could be stuck at 0 for a while, right (or maybe it wouldn't get to the progress bar for a little while--I need to spend some more time understanding into how this works)?

I'm not convinced we need to include those as a part of the whole function's progress (plus I don't know how you'd indicate the two helper functions, percentage-wise), but if we're just logging the actual mutual information calculation, maybe we should be clear about that.

Reply:
Yeah, I was thinking something similar. Maybe since those functions both loop over the columns, we can add those counts to the total expected loop count and update after each one is called. Still won't be perfect, but might be better - and would give a progress update sooner.

Reply:
Yes, that would make sense to me! Is it possible to have separate sections of the progress bar? Like "data preparation" vs "mutual info calculations" that might let us be a little more precise?

Reply:
@tamargrey We will likely add stages name and stage progress in the future. So you get overall progress, and can see the current stage it is on.

Reply:
Right now this is just implementing the ability to call a progress callback function. The progress bar update will come later in another PR - not sure what we will be able to do there yet, as I haven't looked into that yet.